### ISSUE TITLE
Accessibility not work well on TextField

### ISSUE THREAD

**Issue by webbpan0** [flutter/flutter#20718]
For textfield with label "Name", value "John", and count limit "4/20", the talkback order like:
John, Name 4 slash 20, Edit Box

Which is not well constructed, it should be something like:
Text input for Name, current value is John, character count 4 over 20 (or 4 out of 20)

This block product launch of customer: mulligan
Please help to fix it.

**Comment by DaveShuckerow** [comment]
Hi @webbpan0 thanks for filing this issue, can you reach out to me and @jonahwilliams internally to provide more information about this problem and steps to reproduce it inside of the Mulligan codebase?

Thanks!

**Comment by jonahwilliams** [comment]
Deduping this into https://github.com/flutter/flutter/issues/15187
